### DISCUSSION TITLE
getServerSideProps running locally, but not on Vercel

### DISCUSSION THREAD

**Discussion by dsurwolee** [vercel/next.js#44708] [Help]
I have an index.js file within a dynamic router page [lesson]. The SSR function is set up in the index.js file such that when there's a request to the lesson page, the function will fetch file paths to markdown files then parses out the front matter and content and returns it as prop.

**Here's my SSR function:**

```
export async function getServerSideProps({ params: { lesson, course } }) {

    const ssrpropStatus = 'SSR ran!'

    // Fetch the list of markdown lessons in the course folder 
    const files = fs.readdirSync(path.join(process.cwd(), `posts/${course}`))
    const contentPaths = files.map(filename => ({
        lesson: filename.replace('.md',''),
        course: course
    })) 

    // For the selected lesson, convert it to html
    const markdownWithMeta = fs.readFileSync(path.join(process.cwd(), `posts/${course}`, lesson + '.md'), 'utf-8')    
    const { data: frontmatter, content } = matter(markdownWithMeta)

    return { 
        props: {
            contentPaths,
            frontmatter,
            content,
            ssrpropStatus
        },
    }
}
```

**Here's the component that receives the prop:**

```
export default function LessonPage({ contentPaths, frontmatter, content, ssrpropStatus }) {

    // console.log('content', content)
    console.log('contentPaths', contentPaths)
    console.log(ssrpropStatus)

    return (
        <div>
            <SideNav lessonList={contentPaths} />
            <div>
                Content
            </div>
            <div>
                {content}
            </div>
        </div>     
    )
}
```

The page request leads to 500 error on Vercel given that the SideNav component has a .map function that runs through contentPaths. But, because the ServerSideProp is not running, undefined is being passed as prop.

On my local computer, everything works fine. And, I get that, ideally, I should have a try-catch block. But, why is the function not executing on Vercel?

Just to make sure that it's not the code issue within getServerSideProps, I assigned a const variable **ssrpopStatus** and see if this will return 'SSR ran!' as prop. But, this is also returning undefined.

Any help on this would be much appreciated!

**Comment by icyJoseph** [comment]
While I read your snippet, just thought I'd ask, what Next.js version are you using? And, do you have middleware?

**Reply by icyJoseph** [reply]
Btw:

> But, why is the function not executing on Vercel?

This is not correct. The function might be running, but crashing out, and then the runtime handling the task (GSSP is ran pretty much like a API route, on a separate process), sees and error and returns 500. So the try/catch is a must!

**Reply by icyJoseph** [reply]
Alright after reading a bit, I think I'd like to have a try catch and see if the file system calls are working as expected. As said before, GSSP runs as a task, which is why we use the `process.cwd()` call to get access to the directory that contains the actual server and not the task directory.

I'd like to make sure that it is not the file system that's betraying us here.

A couple of follow ups:

- Are you using PWA?
- Next.js version? / Middleware?
- Could it be that your payload exceeds 5Mb?

**Reply by dsurwolee** [reply]
Hey Joseph, thanks for getting back to me. I'm not sure about PWA and middleware (I'm a dev-newbie so bare with me as I try to figure those out one at a time).

I think it's best if I share the Github Repo: https://github.com/dsurwolee/datainterview

I'm using Next.Js version 13.1.1. 

For the payload, possibly, as I'm using serversideprop to fetch markdown files. But, even after disabling it, I'm still having an issue getting the serversideprop function to run.

After further investigation, it seems that the function call isn't working in [<name of slug>].js page directory, but works fine in index.js. I'm not sure why.

I've created this repo: https://github.com/dsurwolee/ssr-test to see if I can replicate the issue.

**Reply by icyJoseph** [reply]
There's no PWA, nor middleware in your application.

Locally, even the production build works fine, so I guess the issue has to be Vercel specific.

I wonder if:

```json
{
    "rewrites":  [
      {"source": "/(.*)", "destination": "/"}
    ]
}
```

Might be the issue?

**Reply by dsurwolee** [reply]
Joseph, you are a magician! After you pointed this out, I tried removing it, and the serversideprop function now works! Thank you so much!

**Reply by icyJoseph** [reply]
Nice! What did you want to do with that configuration anyway? Don't forget to mark as answered, have a good day!
